app/models/bundle.rb

```diff
@@ -5,23 +5,12 @@ class Bundle < Product
 
   def products_for_group_select
     products = facility.products.where.not(type: "Bundle").order(:type, :name)
-    current_group = []
-    current_opts  = []
-    groups = []
-    products.each do |p|
-      if p.class.name != current_group[0]
-        unless current_opts.empty?
-          current_group << current_opts
-          groups << current_group
-          current_group = []
-          current_opts  = []
-        end
-        current_group << p.class.name.pluralize
-      end
-      current_opts << [p.to_s_with_status, p.id]
+    options = {}
+    products.group_by { |p| p.class.name.pluralize }.each do |cname, products|
+      options[cname] ||= []
+      products.map {|p| options[cname] << [p.to_s_with_status, p.id]}
```

Comment:
I think we can invert this somewhat since we're already grouped:
`options[cname] = products.map { |p| [p.to_s_with_status, p.id] }` I think this reads a little clearer, and doesn't need the default anymore.